geopyspark/geotrellis/tile_layer.py

```diff
@@ -233,17 +233,48 @@ def collect_pyramid_metadata(geopysc,
     else:
         output_crs = ""
 
-    result = metadata_wrapper.collectPythonPyramidMetadata(key,
-                                                           java_rdd,
-                                                           schema,
-                                                           crs,
-                                                           tile_size,
-                                                           resolution_threshold,
-                                                           max_zoom,
-                                                           output_crs)
+    result = metadata_wrapper.collectPythonMetadata(key,
+                                                    java_rdd,
+                                                    schema,
+                                                    crs,
+                                                    tile_size,
+                                                    resolution_threshold,
+                                                    max_zoom,
+                                                    output_crs)
 
     return (result._1(), json.loads(result._2()))
 
+def collect_pyramid_floating_metadata(geopysc,
```

Comment:
`pyramid` and `floating` are mutually exclusive. If layer is part of a pyramid, it must be one of the power of two levels with corresponding resolution. Floating layouts are for layouts with resolutions and tile layouts that don't match any pyramid level, but rather match the source data as closely as possible.